botocore/__init__.py

```diff
@@ -70,6 +67,11 @@ def xform_name(name, sep='_', _xform_cache=_xform_cache):
         return name
     key = (name, sep)
     if key not in _xform_cache:
+        if _special_case_transform.search(name) is not None:
+            is_special = _special_case_transform.search(name)
```

Comment:
It's actually not faster for the common case, which is why I reversed the order (you aren't storing a local var).  Definitely a microoptimization though...